app/content/publication-api.md

```diff
@@ -0,0 +1,168 @@
+        <p>Use the <code>refreshToken</code> to generate a new <code>accessToken<code>.</p>
+      </td>
+    </tr>
+    <tr id="revokeButton" class="hidden">
+      <td>
+        <div class="button"></div>
+      </td>
+      <td>
+        <p>Revoke the current <code>accessToken</code> and log out the current user.</p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<div id="GISOutput"></div>
+
+
+
+### Entitlements Response
+
+Access tokens periodically expire but you can [obtain new ones using the Refresh Token ](https://developers.google.com/identity/protocols/oauth2/web-server#offline)without prompting the user for permission (including when the user is not present). As a returning user, you will see above the Refresh Token which was stored. When you click again on SIWG, the corresponding Access Token is used but no new Authorization code is needed.
+
+
+# Implementation Samples
+
+While most of the logic for using the Publication API can be done client-side,
+certain aspects of the OAuth flow must be done server-side. The following code
+samples are split across the client and server, illustrating the type of logic
+that should happen at each layer.
+
+## Client-side code sample
+
+We suggest instantiatiating this flow by calling the GIS SDK's `google.accounts.oauth2.initCodeClient` function in the callback of the SIWG button.
+
+
+```html
+  <head>
+    <script>
+
+      // Exchange authorization code for access+refresh tokens
+      async function exchangeAuthCodeForTokens(authorizationCode) {
+        let url = "https://oauth2.googleapis.com/token?client_id=" + YOUR_OAUTH_CLIENT_ID + "&client_secret=" + YOUR_OAUTH_CLIENT_SECRET + "&code=" + authorizationCode + "&grant_type=authorization_code&redirect_uri=" + REDIRECT_URL
+        var requestOptions = {method: 'POST', redirect: 'follow'};
+        return fetch(url, requestOptions)
+          .then(response => response.json())
+          .catch(error => console.log('error', error));
+      }
+
+      function initGISAuthCodeClientRedirect(hint) { // GIS Authorization client
+        return google.accounts.oauth2.initCodeClient({
+          client_id: YOUR_OAUTH_CLIENT_ID,
+          scope: 'email',
+          hint: hint,
+          ux_mode: 'redirect',
+          redirect_uri: REDIRECT_URL, // Add to list of allowed redirects in Cloud Console
+          state: 'redirect_from_gis_authz_client',
+          select_account: false
+        });
+      }
+
+      // Initialise the GIS Authentication client
+      google.accounts.id.initialize({
+        client_id: YOUR_OAUTH_CLIENT_ID,
+        callback: (response) => {
+          let user_identifier = parseJwt(response.credential).sub
+          let valid_access_token_in_db = getAccessTokenFromStorage(user_identifier)
+          if (valid_access_token_in_db) { // Case 1: known user with valid token
+            queryPublicationAPI(publication_id, valid_access_token_in_db)
+              .then(entitlements => process_entitlements(entitlements))
```

Comment:
Worth calling out that `process_entitlements` needs to be defined by the publisher? Maybe add a placeholder for it at the beginning of the script like we do for EA?  https://developers.google.com/news/subscribe/extended-access/integration-steps/web-implementation?feed=client-side#example-code-3

Reply:
Good call. I added a stub version of the function above.